include/swift/AST/CASTBridging.h

```diff
@@ -31,6 +31,69 @@
 #endif
 
 SWIFT_BEGIN_NULLABILITY_ANNOTATIONS
+
+typedef struct BridgedDeclContext {
+  void *_Nonnull raw;
+} BridgedDeclContext;
+
+// Define the bridging wrappers for each AST node.
+#define AST_BRIDGING_WRAPPER_NONNULL(Name)                                     \
+  typedef struct {                                                             \
+    void *_Nonnull raw;                                                        \
+  } Bridged##Name;
+
+#define AST_BRIDGING_WRAPPER_NULLABLE(Name)                                    \
+  typedef struct {                                                             \
+    void *_Nullable raw;                                                       \
+  } Bridged##Name;
+
+#include "swift/AST/ASTBridgingWrappers.def"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Declare `.asDecl` on each BridgedXXXDecl type.
```

Comment:
Conceptually these are upcasts, right? Might be worth pointing out in the comment.